docs/extensions/readme.md

```diff
@@ -1173,3 +1175,56 @@ An object with an optional property.
   }
 }
 ```
+
+## x-ms-header-collection-prefix
+Use `x-ms-header-collection-prefix` to deal with prefixes for headers. Has different behavior if it refers to a request header or a response header:
+
+- Request header: All keys in the request headers will be prefixed with the prefix value before being sent to the service.
+- Response header: Only response headers that start with the prefix specified here will be returned to users. Additionally, the prefix will be stripped from the response header key before being returned to users
+
+Additionally, applying this extension to a schema forces the schema to become a dictionary.
+
+**Schema**: `string`. Name of the prefix you want to append / filter by. A common value for storage libraries is `x-ms-meta-`.
+
+**Request Example**:
+
+```json
+"parameters": [
+  "x-ms-meta": {
+    "type": "string",
+    "x-ms-client-name": "Metadata",
+    "x-ms-header-collection-prefix": "x-ms-meta-"
+  }
+]
+```
+This request parameter will be forced to be a dictionary schema, and all keys in this dictionary will be prefixed with `x-ms-meta-`.
+So, if you input `{"key": "value"}` through this parameter, `{"x-ms-meta-key": "value"}` is what reaches the service.
+
+**Response Example**:
+
+```json
+"responses": {
+  "200": {
+    "description": "Success",
+    "headers": {
+      "x-ms-meta": {
+        "type": "string",
+        "x-ms-client-name": "Metadata",
+        "x-ms-header-collection-prefix": "x-ms-meta-"
+      }
+    }
+  }
+}
+```
+
+This response header parameter will be forced to be a dictionary schema. Only entries with prefix `x-ms-meta-` will be returned to users,
+and this prefix will be stripped before be returned to users. So if the response from the service is
+
+```json
```

Comment:
I don't think this example helps, since I believe this applies only to Headers, so having an example in JSON it confusing